frontend/awx/resources/inventories/inventoryHostsPage/InventoryHostForm.tsx

```diff
@@ -0,0 +1,195 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { PageHeader, useGetPageUrl } from '../../../../../framework';
+import { useTranslation } from 'react-i18next';
+import { useNavigate, useParams } from 'react-router-dom';
+import {
+  usePageNavigate,
+  PageFormSubmitHandler,
+  PageLayout,
+  PageFormTextInput,
+  PageFormDataEditor,
+} from '../../../../../framework';
+import { requestPatch } from '../../../../common/crud/Data';
+import { usePostRequest } from '../../../../common/crud/usePostRequest';
+import { awxAPI } from '../../../common/api/awx-utils';
+import { AwxPageForm } from '../../../common/AwxPageForm';
+import { AwxHost } from '../../../interfaces/AwxHost';
+import { AwxRoute } from '../../../main/AwxRoutes';
+import { PageFormSection } from '../../../../../framework/PageForm/Utils/PageFormSection';
+import { useEffect, useState } from 'react';
+import { useGetHost } from '../../hosts/hooks/useGetHost';
+import { useGetInventory } from '../InventoryPage/InventoryPage';
+
+export interface IHostInput {
+  name: string;
+  description?: string;
+  variables?: string;
+}
+
+export function CreateHost() {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const pageNavigate = usePageNavigate();
+  const getPageUrl = useGetPageUrl();
+  const params = useParams<{ id: string; inventory_type: string; host_id: string }>();
+  const postRequest = usePostRequest<AwxHost>();
+
+  const onSubmit: PageFormSubmitHandler<IHostInput> = async (hostInput: IHostInput) => {
+    const modifiedHostInput = { ...hostInput, inventory: Number(params.id) };
+    const newHost = await postRequest(awxAPI`/hosts/`, modifiedHostInput as AwxHost);
+    pageNavigate(AwxRoute.InventoryHostDetails, {
+      params: { inventory_type: params.inventory_type, id: params.id, host_id: newHost.id },
+    });
+  };
+
+  const onCancel = () => navigate(-1);
+
+  const inventoryResponse = useGetInventory(params.id, params.inventory_type);
+
+  return (
+    <>
```

Comment:
Maybe use the `PageLayout` component instead of an empty tag to get rid of the scroll in the body of the form